### PR TITLE
#56 filip

### PR DESCRIPTION
**Pull request by davinci-school**

### CODE REVIEW

**Comment by davinci-school** [comment]
https://skoladavinci.tpondemand.com/entity/56

File: user-page/user_homepage/index.html [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    <link rel="stylesheet" href="master.css">
+    <link href="https://fonts.googleapis.com/css?family=Roboto&display=swap" rel="stylesheet">
+    <script src="jquery-3.4.1.min.js"></script>
+</head>
+
+<body>
+    <header>
+        <nav>
+            <h1>Welcome to attendace system!</h1>
+            <button id="menuButton">menu</button>
```
Comment (davinci-school) [inline user-page/user_homepage/index.html]: Do we really want to have a menu for the user? We didn't plan this.

File: user-page/user_homepage/script.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+//get month name based on date
+function getMonthName(data) {
+    var monthList = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+    var month = ""; //set month as empty string
+
+    // go through data and extract numbers that corespond with month
+    //data dor month are on 5th and 6th place, therefore i starts at 5 and ends with 6
+    for (let i = 5; i < 7; i++) {
+        month = month + data.date[i];
+    };
```
Comment (davinci-school) [inline user-page/user_homepage/script.js]: This can be replaced by calling function `getMonthNumber`

File: user-page/user_homepage/script.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    minutesStayed = minutesStayed - hoursStayed * 60;
+    var duration = hoursStayed + ":" + minutesStayed;
+    return duration;
+
+
+};
+
+//add History Log
+function appendHistoryLog(data, divId) {
+    var newDiv = document.createElement("div");
```
Comment (davinci-school) [inline user-page/user_homepage/script.js]: Are you planning to replace the DIV element by table?

File: user-page/user_homepage/script.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
+//fake data used throughout the script for testing
```
Comment (davinci-school) [inline user-page/user_homepage/script.js]: You did a lot of work on your own, formatting the time. I suggest to read [this](https://www.toptal.com/software/definitive-guide-to-datetime-manipulation) when working with time, maybe an existing library would satisfy your needs.

File: user-page/user_homepage/script.js (hunk 3 above)
Comment (fzajic) [reply user-page/user_homepage/script.js]: Not really. With designs i sketched. Divs make more sense. I may use grid in the future. Divs however, can be maintained.

File: user-page/user_homepage/script.js (hunk 4 above)
Comment (fzajic) [reply user-page/user_homepage/script.js]: I actually thought about standardizing time in JS format. This was after I wrote all the functions. I may use this solution in the future.

File: user-page/user_homepage/index.html (hunk 1 above)
Comment (fzajic) [reply user-page/user_homepage/index.html]: Yes we do. I am not planning on full development of dropdown menu. Its just a placeholder for now.

File: user-page/user_homepage/script.js [hunk 5]
```diff
@@ -146,30 +132,37 @@ function appendHistoryLog(data, divId) {
...
+    //after get request is succes, append data to page
+    .then(function() {
+        for (let index = 0; index < 10; index++) {
+            appendHistoryLog(data[index], "session" + index);
+        }
+    })
+    .then(function() {
+        let storedData = sessionStorage.getItem("dataFromJson");
+        storedData = JSON.parse(JSON.stringify(storedData));
+        console.log(storedData);
```
Comment (davinci-school) [inline user-page/user_homepage/script.js]: on line 160 you call the function appendHistoryLog but you didn't convert the data to JSON. You do this at line 165. 
Add line before 158 `data_json = JSON.parse(JSON.stringify(data));` and replace the data on line 160 to data_json

**Pull request merged**